### PR TITLE
Build on Azure pipelines and upload to anaconda.org

### PR DESCRIPTION
**Pull request by TomAugspurger**
The primary motivation here is to change the upload location from the (discontinued) rackspace server to anaconda.org. In the meantime, I switched from travis to azure-pipelines. It's roughly the same as https://github.com/MacPython/scikit-learn-wheels/pull/23.

It's passing on my fork at https://dev.azure.com/pandas-dev/pandas-wheels/_build/results?buildId=29821&view=results

cc @WillAyd, who has been doing similar work on pandas-releases.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
We may not get a CI build here till this is merged.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+schedules:
```
Comment (WillAyd) [inline azure-pipelines.yml]: I think can temporarily remove this block to have it run on a pull request

File: azure/posix.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          setup_test_venv
+          install_run $PLAT
+          teardown_test_venv
+        displayName: Install wheel and test
+        condition: eq(variables['SKIP_BUILD'], 'false')
+
+      - task: PublishTestResults@2
+        inputs:
+          testResultsFiles: '$(TEST_DIR)/$(JUNITXML)'
+          testRunTitle: ${{ format('{0}-$(Agent.JobName)', parameters.name) }}
```
Comment (thomasjpfan) [inline azure/posix.yml]: This can be removed if you do not wish to upload the test results.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          NIGHTLY_BUILD: "true"
+        py_3.8_64:
+          MB_PYTHON_VERSION: "3.8"
+          NP_BUILD_DEP: "numpy==1.17.3"
+          NIGHTLY_BUILD: "true"
+
+  - template: azure/posix.yml
+    parameters:
+      name: macOS
+      vmImage: xcode9-macos10.13
```
Comment (thomasjpfan) [inline azure-pipelines.yml]: To make azure happy

```suggestion
      vmImage: macOS-10.14
```

**Comment by TomAugspurger** [comment]
OK https://dev.azure.com/pandas-dev/pandas-wheels/_build?definitionId=2&_a=summary passed. Going to merge this now, and then will probably have a few followup things to do.

**Pull request merged**